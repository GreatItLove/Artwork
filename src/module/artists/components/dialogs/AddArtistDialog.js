import React from 'react';
import Dialog from 'material-ui/Dialog';
import PropTypes from 'prop-types';

import AddArtist from '../../containers/AddArtist';

const propTypes = {
  adding: PropTypes.bool.isRequired,
  closeAddDialog: PropTypes.func.isRequired,
};

/**
 * @function AddArtist
 * @param {Boolean} adding
 * @param {Function} closeAddDialog
 * @returns {XML}
 */
function AddArtistDialog({adding, closeAddDialog}) {
  return (
    <Dialog
      onRequestClose={closeAddDialog}
      open={adding}
      bodyStyle={{overflowY: 'unset'}}
      overlayStyle={{backgroundColor: '#797979', opacity: '0.8'}}
      paperProps={{style: {backgroundColor: 'none'}, zDepth: 0}}>
      {adding && <AddArtist closeDialog={closeAddDialog} />}
    </Dialog>
  );
}

AddArtistDialog.propTypes = propTypes;

export default AddArtistDialog;
