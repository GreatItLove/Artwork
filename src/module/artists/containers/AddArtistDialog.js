import {connect} from 'react-redux';

import AddArtistDialog from '../components/dialogs/AddArtistDialog';
import {closeArtistAddDialog} from '../artistsActions';

const mapStateToProps = state => ({
  adding: state.artists.adding,
});

const mapDispatchToProps = dispatch => ({
  closeAddDialog: () => {
    dispatch(closeArtistAddDialog());
  },
});

export default connect(mapStateToProps, mapDispatchToProps)(AddArtistDialog);
