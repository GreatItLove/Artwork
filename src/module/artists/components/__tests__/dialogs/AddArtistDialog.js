import React from 'react';
import {shallow} from 'enzyme';
import toJSON from 'enzyme-to-json';

import AddArtistDialog from '../../dialogs/AddArtistDialog';

describe('AddArtistDialog', () => {
  const props = {
    adding: true,
    closeAddDialog: jest.fn(),
  };
  it('Snapshot of AddArtistDialog when adding is true', () => {
    const wrapper = shallow(<AddArtistDialog {...props} />);
    expect(toJSON(wrapper)).toMatchSnapshot();
  });
  it('Snapshot of AddUserDialog whan adding is false ', () => {
    const propsAddingFalse = {...props, ...{adding: false}};
    const wrapper = shallow(<AddArtistDialog {...propsAddingFalse} />);
    expect(toJSON(wrapper)).toMatchSnapshot();
  });
});
