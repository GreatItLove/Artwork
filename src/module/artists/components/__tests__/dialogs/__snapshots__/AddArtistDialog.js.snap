// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AddArtistDialog Snapshot of AddArtistDialog when adding is true 1`] = `
<Dialog
  autoDetectWindowHeight={true}
  autoScrollBodyContent={false}
  bodyStyle={
    Object {
      "overflowY": "unset",
    }
  }
  modal={false}
  onRequestClose={[Function]}
  open={true}
  overlayStyle={
    Object {
      "backgroundColor": "#797979",
      "opacity": "0.8",
    }
  }
  paperProps={
    Object {
      "style": Object {
        "backgroundColor": "none",
      },
      "zDepth": 0,
    }
  }
  repositionOnUpdate={true}
>
  <Translate(Connect(ReduxForm))
    closeDialog={[Function]}
  />
</Dialog>
`;

exports[`AddArtistDialog Snapshot of AddUserDialog whan adding is false  1`] = `
<Dialog
  autoDetectWindowHeight={true}
  autoScrollBodyContent={false}
  bodyStyle={
    Object {
      "overflowY": "unset",
    }
  }
  modal={false}
  onRequestClose={[Function]}
  open={false}
  overlayStyle={
    Object {
      "backgroundColor": "#797979",
      "opacity": "0.8",
    }
  }
  paperProps={
    Object {
      "style": Object {
        "backgroundColor": "none",
      },
      "zDepth": 0,
    }
  }
  repositionOnUpdate={true}
/>
`;
